packages/gestalt/src/IconButtonLink.js

```diff
@@ -0,0 +1,214 @@
+   * SVG icon from the Gestalt icon library to use within Icon.
+   *
+   * See the [icon library](https://gestalt.pinterest.systems/foundations/iconography/library) to explore available options.
+   */
+  icon?: $Keys<typeof icons>,
+  /**
+   * Primary color to apply to the [Icon](/web/icon).
+   */
+  iconColor?: 'gray' | 'darkGray' | 'red' | 'white' | 'brandPrimary',
+  /**
+   * Callback fired when the component is clicked, pressed or tapped. See [GlobalEventsHandlerProvider](/web/utilities/globaleventshandlerprovider#Link-handlers) to learn more about link navigation.
+   */
+  onClick?: ({|
+    event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+  /**
+   * Sets a padding for the IconButtonLink.
+   */
+  padding?: 1 | 2 | 3 | 4 | 5,
+  /**
+   * Specifies the relationship between the current document and the linked document.
+   */
+  rel?: 'none' | 'nofollow',
+  /**
+   * The maximum height and width of IconButtonLink.
+   */
+  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
+  /**
+   * Removes IconButtonLink from sequential keyboard navigation to improve accessibility.
+   */
+  tabIndex?: -1 | 0,
+  /**
+   * Define the frame or window to open the anchor defined on \`href\`.
+   */
+  target?: null | 'self' | 'blank',
+  /**
+   * Adds a [Tooltip](/web/tooltip) on hover/focus of the IconButtonLink.
+   */
+  tooltip?: {|
+    accessibilityLabel?: string,
+    inline?: boolean,
+    idealDirection?: 'up' | 'right' | 'down' | 'left',
+    text: string,
+    zIndex?: Indexable,
+  |},
+|};
+
+/**
+ * [IconButtonLink](https://gestalt.pinterest.systems/web/iconbuttonlink) is mainly used as navigational element.
+ *
+ * ![IconButton light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/IconButton.spec.mjs-snapshots/IconButton-chromium-darwin.png)
+ * ![IconButton dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/IconButton-dark.spec.mjs-snapshots/IconButton-dark-chromium-darwin.png)
+ *
+ */
+const IconButtonLinkWithForwardRef: AbstractComponent<Props, HTMLAnchorElement> = forwardRef<
+  Props,
+  HTMLAnchorElement,
+>(function IconButtonLink(
+  {
+    href,
+    rel,
+    target,
+    accessibilityLabel,
+    bgColor,
+    dangerouslySetSvgPath,
+    dataTestId,
+    disabled,
+    icon,
+    iconColor,
+    onClick,
+    padding,
+    tabIndex = 0,
+    tooltip,
+    size = 'lg',
+  }: Props,
+  ref,
+): Node {
+  const innerRef = useRef<null | HTMLAnchorElement>(null);
+
+  // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+  // that renders <IconButton ref={inputRef} /> to call inputRef.current.focus()
+  useImperativeHandle(ref, () => innerRef.current);
+
+  const [isActive, setActive] = useState(false);
+  const [isFocused, setFocused] = useState(false);
+  const [isHovered, setHovered] = useState(false);
```

Comment:
why not `useInteractiveStates`?

Reply:
Good catch, I didn't know about this hook. Much cleana'